File: src/main/java/com/pedigree/model/Relationship.java
package com.pedigree.model;

public class Relationship {
    public enum Type { SPOUSE_TO_FAMILY, FAMILY_TO_CHILD }

    private Type type;
    private String fromId;
    private String toId;

    public Type getType() { return type; }
    public void setType(Type type) { this.type = type; com.pedigree.util.DirtyFlag.setModified(); }
    public String getFromId() { return fromId; }
    public void setFromId(String fromId) { this.fromId = fromId; com.pedigree.util.DirtyFlag.setModified(); }
    public String getToId() { return toId; }
    public void setToId(String toId) { this.toId = toId; com.pedigree.util.DirtyFlag.setModified(); }
}

File: src/main/kotlin/com/family/tree/model/Relationship.kt
package com.family.tree.model

import com.family.tree.util.DirtyFlag

class Relationship {
    enum class Type { SPOUSE_TO_FAMILY, FAMILY_TO_CHILD }

    var type: Type? = null
        set(value) { field = value; DirtyFlag.setModified() }
    var fromId: String? = null
        set(value) { field = value; DirtyFlag.setModified() }
    var toId: String? = null
        set(value) { field = value; DirtyFlag.setModified() }
}
